File: Java/Sorting Algorithms/BubbleSort.java
import java.util.*;
public class BubbleSort {
	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);
		System.out.println("Enter Array size : ");
		int n = sc.nextInt();
		int a[] = new int[n];
		for (int i = 0; i < n; i++) {
			System.out.println("Enter Number at " + i + " th Position : ");
			a[i] = sc.nextInt();
		}
		int temp;
		for (int i = 0 ; i < a.length; i++) {
			for (int j = 0; j < a.length - i - 1; j++) {
				if (a[j] > a[j + 1]) {
					temp = a[j];
					a[j] = a[j + 1];
					a[j + 1] = temp;
				}
			}
		}
		for (int i = 0 ; i < a.length ; i++) {
			System.out.print(" " + a[i]);
		}
	}
}

File: C/Array/BubbleSort.c
#include<stdio.h>

int main (){
int a[10],size,temp,o;
printf("ENTER ARRAY SIZE ");
scanf("%d",&size);
for(int i=0;i<size;i++){
printf("Enter Array At %d th Index",i);
scanf("%d",&a[i]);

}
for(int i=0;i<size-1;i++){
printf("Pass -%d\n",i+1);
for(int j=0;j<size-i-1;j++){
if(a[j]>a[j+1]){
temp=a[j];
a[j]=a[j+1];
a[j+1]=temp;
}
printf("[ ");
for(int i=0;i<size;i++){
printf("%d ",a[i]);
}
printf(" ] ");
printf("\n");
printf("\n");
}
o=i;
}
printf("Number of Passes %d",o+1);
return 0;
}
